boefjes/plugins/kat_nmap_ports/main.py

```diff
@@ -0,0 +1,64 @@
+import re
+
+from os import getenv
+from ipaddress import ip_address, IPv6Address
+from typing import List, Tuple, Union
+
+import docker
+
+from boefjes.job_models import BoefjeMeta
+
+NMAP_IMAGE = "instrumentisto/nmap:latest"
+
+
+def run_nmap(args: List[str]) -> str:
+    """Run Nmap in Docker."""
+    client = docker.from_env()
+    return client.containers.run(NMAP_IMAGE, args, remove=True).decode()
+
+
+def build_nmap_arguments(host: str, ports: str) -> str:
+    """Build nmap arguments from the hosts IP with the required ports."""
+    ip = ip_address(host)
+    args = ["nmap", "-T4", "-Pn", "-r", "-v10", "-sV", "-sS", "-sU", f"-p{ports}"]
+    if isinstance(ip, IPv6Address):
+        args.append("-6")
+    args.extend(["-oX", "-", host])
+
+    return args
+
+
+def get_ports(
```

Comment:
Two suggestions:
- Introduce a constant for this pattern
- Add `ports` as an argument; this makes the function (better) testable

Reply:
I don't think the `getenv("PORTS")` will be working when introduced in the function argument, will it? (I.e. that will introduce the same problem as solved in https://github.com/minvws/nl-kat-boefjes/pull/28). Or would you prefer something like `ports=None` and then in the function `if ports is None: ports = getenv("PORTS")`?

Reply:
I meant getting the ports outside this function and passing it as an argument. Because the current implementation introduces a dependency due to `getenv`. Injecting this dependency (by passing a list of ports) makes the function less coupled and improves cohesion (and even more important, testing/ reproducibility). Your suggestion would already improve this 👍 
```python3
ports = getenv(...)
get_ports(ports)
```

Reply:
Okay, is the current version more to your liking?

Reply:
I think it looks great 👌